sdk/trace/src/main/scala/org/typelevel/otel4s/sdk/trace/SdkSpanBackend.scala

```diff
@@ -0,0 +1,277 @@
+/*
+ * Copyright 2023 Typelevel
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.typelevel.otel4s.sdk
+package trace
+
+import cats.Applicative
+import cats.Monad
+import cats.effect.Clock
+import cats.effect.Concurrent
+import cats.effect.Ref
+import cats.syntax.flatMap._
+import cats.syntax.functor._
+import cats.syntax.semigroup._
+import org.typelevel.otel4s.Attribute
+import org.typelevel.otel4s.AttributeKey
+import org.typelevel.otel4s.meta.InstrumentMeta
+import org.typelevel.otel4s.sdk.common.InstrumentationScopeInfo
+import org.typelevel.otel4s.sdk.trace.data.EventData
+import org.typelevel.otel4s.sdk.trace.data.LinkData
+import org.typelevel.otel4s.sdk.trace.data.SpanData
+import org.typelevel.otel4s.sdk.trace.data.StatusData
+import org.typelevel.otel4s.trace.Span
+import org.typelevel.otel4s.trace.SpanContext
+import org.typelevel.otel4s.trace.SpanKind
+import org.typelevel.otel4s.trace.Status
+
+import scala.concurrent.duration.FiniteDuration
+
+final class SdkSpanBackend[F[_]: Monad: Clock] private (
+    spanLimits: SpanLimits,
+    spanProcessor: SpanProcessor[F],
+    immutableState: SdkSpanBackend.ImmutableState,
+    mutableState: Ref[
+      F,
+      SdkSpanBackend.MutableState
+    ] // todo: use AtomicCell[F, SdkSpanBackend.MutableState]
+) extends Span.Backend[F]
+    with ReadWriteSpan[F] {
+
+  def meta: InstrumentMeta[F] = InstrumentMeta.enabled
+
+  def context: SpanContext = immutableState.context
+
+  // todo: apply attribute span limits
+  def addAttributes(attributes: Attribute[_]*): F[Unit] =
+    if (attributes.isEmpty) Applicative[F].unit
+    else
+      mutableState.update { s =>
+        s.copy(attributes = s.attributes |+| Attributes(attributes: _*))
```

Comment:
The `spanLimits` should be applied here. 
To make it work, we should also tweak the implementation of the `Attributes` class.